### PR TITLE
chore: add error handling to `finishRun()`

### PR DESCRIPTION
**Pull request by Zidious**
Add error handling to `finishRun()`:

- [ ] playwright
- [ ] puppeteer
- [ ] webdriverio
- [ ] webdriverjs

### CODE REVIEW

File: packages/playwright/src/index.ts [hunk 1]
```diff
@@ -147,7 +147,18 @@ export default class AxeBuilder {
...
+
+    try {
+      return this.finishRun(partials);
+    } catch (error) {
+      throw new Error(
+        `Error: ${JSON.stringify(
+          error,
+          null,
+          2
+        )}\n Please check out https://github.com/dequelabs/axe-core-npm/blob/develop/packages/playwright/error-handling.md`
```
Comment (stephenmathieson) [inline packages/playwright/src/index.ts:159]: I don't think this does what you expect it to.

```
throw new Error(`Error: ${JSON.stringify(new Error("boom"), null, 2)}\n Please ...`)
VM335:1 Uncaught Error: Error: {}
 Please ...
    at <anonymous>:1:7
```

Note `Error: {}`, not `Error: boom`.

File: packages/playwright/src/index.ts [hunk 2]
```diff
@@ -259,6 +270,9 @@ export default class AxeBuilder {
         partialResults,
         options
       })
+      .catch(e => {
+        return e;
```
Comment (stephenmathieson) [inline packages/playwright/src/index.ts:274]: Why `return` the error? Doesn't this mean it's "dropped"?

Comment (stephenmathieson) [reply packages/playwright/src/index.ts:274]: This means the Promise resolves with an instance of an `Error`, thus breaking _all_ type contracts.

```
var result = await new Promise((resolve, reject) => reject(new Error('boom'))).catch(error => { return error })
result instanceof Error
// => true
result
// => Error: boom
    at result (<anonymous>:1:60)
    at new Promise (<anonymous>)
    at <anonymous>:1:20
```

**Review by stephenmathieson — CHANGES_REQUESTED** [review]
`JSON.stringify()`ing an error doesn't do what you're expect it to. Also `return`ing errors breaks type contracts.

File: packages/playwright/src/index.ts (hunk 1 above)
Comment (Zidious) [reply packages/playwright/src/index.ts:159]: Thoughts on simply throwing the `error` and providing the link to the docs. 


```
   try {
      return this.finishRun(partials);
    } catch (error) {
      throw new Error(
        `Error: ${error}\n Please check out https://github.com/dequelabs/axe-core-npm/blob/develop/packages/playwright/error-handling.md`
      );
    }
 ```

File: packages/playwright/src/index.ts [hunk 3]
```diff
@@ -147,7 +147,14 @@ export default class AxeBuilder {
       context
     );
     const partials = await partialResults.getPartials();
-    return this.finishRun(partials);
+
+    try {
+      return await this.finishRun(partials);
+    } catch (error) {
+      throw new Error(
+        `Error: ${error}\n Please check out https://github.com/dequelabs/axe-core-npm/blob/develop/packages/playwright/error-handling.md`
```
Comment (WilcoFiers) [inline packages/playwright/src/index.ts:155]: This won't serialise properly. If you want just the message use error.message, or if you want the stack trace use error.trace.

File: packages/puppeteer/test/axePuppeteer.test.ts [hunk 4]
```diff
@@ -694,6 +694,21 @@ describe('AxePuppeteer', function () {
     });
   });
 
+  describe('axe.finishRun errors', () => {
```
Comment (WilcoFiers) [inline packages/puppeteer/test/axePuppeteer.test.ts:697]: Can you adopt the tests as written?

https://github.com/dequelabs/axe-test-fixtures/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R304

File: packages/webdriverio/error-handling.md [hunk 5]
```diff
@@ -0,0 +1 @@
+# Error Handling
```
Comment (WilcoFiers) [inline packages/webdriverio/error-handling.md:1]: Where's the rest of it?

**Review by WilcoFiers — CHANGES_REQUESTED** [review]
Why didn't you adopt this part of the proposal:
https://github.com/dequelabs/axe-api-team-proposals/pull/204/files#diff-96a02d97446db7824b84e655002469cbb37be2a86b4446e528b368ee3b2e363eR73

File: packages/playwright/error-handling.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# Error Handling
```
Comment (stephenmathieson) [inline packages/playwright/error-handling.md:1]: This is a lovely document 😆

File: packages/playwright/src/index.ts (hunk 3 above)
Comment (stephenmathieson) [inline packages/playwright/src/index.ts:155]: Nitpick - prefixing the error message with "Error:" is redundant.

```
> throw new Error('Error: blah blah blah\nPlease check out https://......')
Uncaught Error: Error: blah blah blah
Please check out https://......
>
```

This now reads _Uncaught Error: Error_. Perhaps remove the leading "Error:" text, as it's already provided by the JS engine?


```suggestion
        `${error}\n Please check out https://github.com/dequelabs/axe-core-npm/blob/develop/packages/playwright/error-handling.md`
```

Comment (stephenmathieson) [inline packages/playwright/src/index.ts:155]: I do not recall what the spec says here. Does `Error#toString()` _always_ return `error.message`? If not, we might want to be explicit here:

```suggestion
        `Error: ${error.message}\n Please check out https://github.com/dequelabs/axe-core-npm/blob/develop/packages/playwright/error-handling.md`
```

Comment (stephenmathieson) [reply packages/playwright/src/index.ts:155]: `error.trace` is not a standard property. Do not use that.

**Pull request merged**